src/node_report.cc

```diff
@@ -439,17 +441,23 @@ static void PrintNetworkInterfaceInfo(JSONWriter* writer) {
 // Report the JavaScript stack.
 static void PrintJavaScriptStack(JSONWriter* writer,
                                  Isolate* isolate,
-                                 Local<String> stackstr,
+                                 Local<Object> error,
                                  const char* trigger) {
-  writer->json_objectstart("javascriptStack");
+  Local<Context> context(isolate->GetCurrentContext());
+  Local<Value> stackstr;
 
-  std::string ss;
+  writer->json_objectstart("javascriptStack");
+  std::string ss = "";
   if ((!strcmp(trigger, "FatalError")) ||
       (!strcmp(trigger, "Signal"))) {
     ss = "No stack.\nUnavailable.\n";
-  } else {
+  } else if (error
+                 ->Get(context,
+                       String::NewFromUtf8(isolate, "stack").ToLocalChecked())
+                 .ToLocal(&stackstr)) {
```

Comment:
What if there’s an exception?

Reply:
`ToLocal` will not throw an error if not exist, only boolean returns

https://v8docs.nodesource.com/node-13.2/d8/d7d/classv8_1_1_maybe_local.html#affda80ff2cf83d6151d5af2fc89976ab

Reply:
@Himself65 `.ToLocal()` will return `false` if `Get()` returns an empty `MaybeLocal`; and V8 APIs return empty `MaybeLocal`s if there is an exception pending, see https://v8docs.nodesource.com/node-13.2/d8/d7d/classv8_1_1_maybe_local.html#details.

Reply:
I’ve mentioned it above but this `Get()` call should also happen inside a `TryCatch` so as not to trigger an uncaught exception

I know this is annoying, but we should really be careful here because this itself can be run as exception-handling code